packages/firestore/src/model/mutation.ts

```diff
@@ -457,29 +515,55 @@ function applySetMutationToRemoteDocument(
   maybeDoc: MaybeDocument | null,
   mutationResult: MutationResult
 ): Document {
-  debugAssert(
-    mutationResult.transformResults == null,
-    'Transform results received by SetMutation.'
-  );
-
   // Unlike applySetMutationToLocalView, if we're applying a mutation to a
   // remote document the server has accepted the mutation so the precondition
   // must have held.
-  return new Document(mutation.key, mutationResult.version, mutation.value, {
+  let newData = mutation.value;
+  if (mutation.fieldTransforms && mutationResult.transformResults) {
```

Comment:
Shouldn't we always either have or not have `mutation.fieldTransforms` and `mutationResult.transformResults`? The transform results for some of the operations are only partial but the entry still exists (e.g array union) .

FYI, if you initialize `mutation.fieldTransforms` to an empty array then this if-check goes away.

Reply:
The `mutation.fieldTransforms` check still shows up for me.

Reply:
Oops, removed them everywhere but here.